buku.py

```diff
@@ -1239,6 +1239,26 @@ def search_by_tag(self, tags):
         self.cur.execute(query, tuple(tags, ))
         return self.cur.fetchall()
 
+
+    def search_any_keyword_and_filter_by_tags(self, sany, deep, stag):
```

Comment:
Instead of a separate API we can do this under each case sany, sall and sreg if args.stag is not None and len(args.stag) > 0.